db/db.go

```diff
@@ -52,13 +93,28 @@ type Loader struct {
 	tracer logging.Tracer
 
 	testTx *TestTx // used for testing: if non-nil, 'loader.BeginTx()' will return this object instead of a real *sql.Tx
+
+	// async flush
+	mu                 sync.Mutex
+	cond               *sync.Cond
+	activeFlushes      int
```

Comment:
you can also use `atomic.Int32` here that way you won't need to lock for reading/writing that field